### PR TITLE
test,docs: Vagrant misc updates for net-next.

### PR DESCRIPTION
**Pull request by brb**
This PR:

- Prevents Vagrant from printing to stdout about running on net-next.
- Adds a note about running `ubuntu-next` on vbox which vsn != 5.2.0.

After merging this PR, we can close #7255 without merging it.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7338)

### CODE REVIEW

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-docs-please

File: Documentation/contributing.rst [hunk 1]
```diff
@@ -661,6 +661,30 @@ supported version of Kubernetes, run the test suite with the following format:
...
+   VirtualBox which version does not match a version of the VM image
+   VirtualBox Guest Additions, Vagrant will install a new version of
+   the Additions with ``mount.vboxsf``. The latter is not compatible with
+   ``vboxsf.ko`` shipped within the VM image, and thus syncing of shared
+   folders will not work.
+
+   To avoid this, one can prevent Vagrant from installing the Additions by
+   putting the following into ``$HOME/.vagrant.d/Vagrantfile``:
+
+   .. code:: bash
```
Comment (ianvernon) [inline Documentation/contributing.rst]: Nit - this isn't bash - is there a `ruby` option?

**Review by ianvernon — CHANGES_REQUESTED** [review]

**Review by joestringer — APPROVED** [review]

File: Documentation/contributing.rst (hunk 1 above)
Comment (joestringer) [reply Documentation/contributing.rst]: Most likely you can just substitute `ruby` here.

Comment (brb) [reply Documentation/contributing.rst]: I didn't want to enable syntax highlighting for the snippet. Anyway, happy to substitute. UPDATE: actually, I didn't highlight anything.

**Comment by brb** [comment]
test-docs-please

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
Test failed due to unrelated to my changes flake.

**Review by joestringer — APPROVED** [review]

**Comment by joestringer** [comment]
@brb in future would you mind looking for a corresponding issue and linking to it, or creating a new issue for it? If we don't document flaky issues, then there's no central place to co-ordinate it and it's difficult to track how often the same issue is being hit.

For this issue I've filed it here:
https://github.com/cilium/cilium/issues/7358

**Comment by ianvernon** [comment]
@be right back  per comments by @aanm in https://github.com/cilium/cilium/pull/7401 , removing needs-backport/1.3 label.

**Pull request merged**